lib/charms/vault_k8s/v0/vault_tls.py

```diff
@@ -123,56 +164,49 @@ def _on_tls_certificates_access_relation_broken(self, event: RelationBrokenEvent
         and reloads vault.
         """
         self._remove_all_certs_from_workload()
-        if not self.charm._ingress_address:
-            return
-        self._generate_self_signed_certs(self.charm._ingress_address)
-        tlslogger.info(
-            "Saved Vault generated CA and self signed certificate to %s.",
-            self.charm.unit.name,
-        )
-        self._reload_vault_container()
+        self.charm.on.config_changed.emit()
 
-    def configure_certificates(self, ingress_address: str) -> None:
+    def configure_certificates(self, subject_ip: str) -> None:
         """Configures the certificates that are used to connect to and communicate with Vault.
 
         Args:
-            ingress_address: The ip address for which the certificates will be configured for.
+            subject_ip: The ip address for which the certificates will be configured for.
         """
-        if self.charm.unit.is_leader() and not self.ca_certificate_set_in_peer_relation():
-            ca_private_key, ca_certificate = generate_vault_ca_certificate()
-            self._set_ca_certificate_secret_in_peer_relation(ca_private_key, ca_certificate)
-            tlslogger.info("Saved the Vault generated CA cert in juju secrets.")
-
-        if not self._tls_file_pushed_to_workload(File.CA) or not self._tls_file_pushed_to_workload(
-            File.CERT
-        ):
-            self._generate_self_signed_certs(ingress_address)
-            tlslogger.info(
-                "Saved Vault generated CA and self signed certificate to %s.",
-                self.charm.unit.name,
-            )
-            self._reload_vault_container()
+        if not self.charm.model.get_relation(TLS_CERTIFICATE_ACCESS_RELATION_NAME):
+            if self.charm.unit.is_leader() and not self.ca_certificate_secret_exists():
+                ca_private_key, ca_certificate = generate_vault_ca_certificate()
+                self._set_ca_certificate_secret(ca_private_key, ca_certificate)
+                tls_logger.info("Saved the Vault generated CA cert in juju secrets.")
+            if not self._tls_file_pushed_to_workload(
+                File.CA
+            ) or not self._tls_file_pushed_to_workload(File.CERT):
+                self._generate_self_signed_certs(subject_ip)
+                tls_logger.info(
+                    "Saved Vault generated CA and self signed certificate to %s.",
+                    self.charm.unit.name,
+                )
+                self._reload_vault()
 
-        elif self.charm.model.get_relation(TLS_CERTIFICATE_ACCESS_RELATION_NAME):
+        else:
```

Comment:
You can just remove the else here and return after reloading since we don't do anything after the else.